File: ios/YGOAPI/YGOAPI/YGORequest.m
//
//  YGORequest.m
//  YGOAPI
//
//  Created by rarnu on 2018/6/30.
//  Copyright © 2018 rarnu. All rights reserved.
//

#import "YGORequest.h"

@implementation YGORequest

+(NSString*) request:(NSString*) url {
    NSString* ret = @"";
    url = [url stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
    @try {
        NSURL * u = [NSURL URLWithString:url];
        NSMutableURLRequest * req = [NSMutableURLRequest requestWithURL:u cachePolicy:0 timeoutInterval:2.0f];
        NSHTTPURLResponse * resp;
        NSData * retData =[NSURLConnection sendSynchronousRequest:req returningResponse:&resp error:nil];
        if (resp.statusCode == 200) {
            ret = [[NSString alloc] initWithData:retData encoding:NSUTF8StringEncoding];
        }
    } @catch (NSException *exception) {
        
    } @finally {
        
    }
    return ret;
}

+(NSString*) search:(NSString*) key page:(NSInteger)page {
    NSString* urlStr =[BASE_URL stringByAppendingFormat:@"/search/%@/%d", key, page];
    return [YGORequest request:urlStr];
}

+(NSString*) cardDetail:(NSString*) hashid {
    NSString* urlStr = [BASE_URL stringByAppendingFormat:@"/card/%@", hashid];
    return [YGORequest request:urlStr];
}

+(NSString*) cardWiki:(NSString*) hashid {
    NSString* urlStr = [BASE_URL stringByAppendingFormat:@"/card/%@/wiki", hashid];
    return [YGORequest request:urlStr];
}

+(NSString*) limit {
    NSString* urlStr = [BASE_URL stringByAppendingString:@"/Limit-Latest"];
    return [YGORequest request:urlStr];
}

+(NSString*) packageList {
    NSString* urlStr = [BASE_URL stringByAppendingString:@"/package"];
    return [YGORequest request:urlStr];
}

+(NSString*) packageDetail:(NSString*) url {
    NSString* urlStr = [BASE_URL stringByAppendingString:url];
    return [YGORequest request:urlStr];
}

+(NSString*) hotest {
    return [YGORequest request:BASE_URL];
}

@end

File: pc/ygoapi/ygorequest.pas
unit ygorequest;

{$mode objfpc}{$H+}

interface

uses
  Classes, SysUtils, fphttpclient;

type

  { TYGORequest }

  TYGORequest = class
  private
    class function request(aurl: string): string;
  public
    class function search(akey: string; apage: Integer): string;
    class function cardDetail(ahashid: string): string;
    class function cardWiki(ahashid: string): string;
    class function limit(): string;
    class function packageList(): string;
    class function packageDetail(aurl: string): string;
    class function hotest(): string;
  end;

implementation

const
  BASE_URL = 'https://www.ourocg.cn';

{ TYGORequest }

class function TYGORequest.request(aurl: string): string;
var
  ret: string = '';
begin
  with TFPHTTPClient.Create(nil) do begin
    try
      ret := Get(aurl);
    except
    end;
    Free;
  end;
  Exit(ret);
end;

class function TYGORequest.search(akey: string; apage: Integer): string;
var
  urlstr: string;
begin
  urlstr:= BASE_URL + Format('/search/%s/%d', [akey, apage]);
  Exit(request(urlstr));
end;

class function TYGORequest.cardDetail(ahashid: string): string;
var
  urlstr: string;
begin
  urlstr:= BASE_URL + Format('/card/%s', [ahashid]);
  Exit(request(urlstr));
end;

class function TYGORequest.cardWiki(ahashid: string): string;
var
  urlstr: string;
begin
  urlstr := BASE_URL + Format('/card/%s/wiki', [ahashid]);
  Exit(request(urlstr));
end;

class function TYGORequest.limit(): string;
var
  urlstr: string;
begin
  urlstr:= BASE_URL + '/Limit-Latest';
  Exit(request(urlstr));
end;

class function TYGORequest.packageList(): string;
var
  urlstr: string;
begin
  urlstr:= BASE_URL + '/package';
  Exit(request(urlstr));
end;

class function TYGORequest.packageDetail(aurl: string): string;
var
  urlstr: string;
begin
  urlstr:= BASE_URL + aurl;
  Exit(request(urlstr));
end;

class function TYGORequest.hotest(): string;
begin
  Exit(request(BASE_URL));
end;

end.

